cli/cli-client/src/main/java/org/infinispan/cli/commands/Migrate.java

```diff
@@ -0,0 +1,148 @@
+package org.infinispan.cli.commands;
+
+import org.aesh.command.Command;
+import org.aesh.command.CommandDefinition;
+import org.aesh.command.CommandException;
+import org.aesh.command.CommandResult;
+import org.aesh.command.GroupCommandDefinition;
+import org.aesh.command.option.Option;
+import org.infinispan.cli.activators.ConnectionActivator;
+import org.infinispan.cli.activators.DisabledActivator;
+import org.infinispan.cli.completers.CacheCompleter;
+import org.infinispan.cli.impl.ContextAwareCommandInvocation;
+import org.kohsuke.MetaInfServices;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 11.0
+ **/
+@MetaInfServices(Command.class)
+@GroupCommandDefinition(name = Migrate.CMD, description = "Performs data migration", groupCommands = {Migrate.Cluster.class})
```

Comment:
If this going to be the home of `migrate configuration` etc, the description should probably be just "Migration operations" similar to the Patch command.